include/pmacc/particles/memory/boxes/ExchangePushDataBox.hpp

```diff
@@ -42,12 +42,12 @@ namespace pmacc
 
         HDINLINE ExchangePushDataBox(
             VALUE* data,
-            TYPE* currentSizePointer,
+            TYPE* particleCount,
             TYPE maxSize,
             PushDataBox<TYPE, PushType> virtualMemory)
             : DataBox<PitchedBox<VALUE, DIM1>>(PitchedBox<VALUE, DIM1>(data))
             , virtualMemory(virtualMemory)
-            , currentSizePointer(currentSizePointer)
+            , parCount(particleCount)
```

Comment:
Good renaming. I can see how you tried to use a different name here but why not simply call the internal variable `particleCount`, too? Would be much nicer to read later on and doesn't suffer from translation problems like me thinking of "parameter count".

Reply:
This would be throw warning that we have shadowing the internal variable. This is also a common source of mistakes.
Typical we use `m_*` for internal variables to avoid it. And use a nice name for the user interface.

Reply:
Yes, I'm aware of this but given that we are dealing with an empty constructor this trade-off might be acceptable.

Reply:
I will change it to `m_particleCount` and also the next line where we have the shadowing issue.